extra/book-template.html

```diff
@@ -83,12 +83,39 @@ <h1 class="title">$title$</h1>
 </header>
 $endif$
 $if(toc)$
-<nav id="$idprefix$TOC" role="doc-toc">
-$if(toc-title)$
-<h2 id="$idprefix$toc-title">$toc-title$</h2>
-$endif$
-$table-of-contents$
-</nav>
+<div>
+  <button id="main-toc-button" class="toc-button" onclick="toggleTOC()">Table of Contents</button>
+  <nav style="display: none;" id="$idprefix$TOC" role="doc-toc">
+  $if(toc-title)$
+  <h2 id="$idprefix$toc-title">$toc-title$</h2>
+  $endif$
+  $table-of-contents$
+  </nav>
+  <script>
+    function toggleTOC() {
+      var toc = document.getElementById('$idprefix$TOC');
+      if (toc.style.display == 'none') {
+        toc.style.display = 'block';
+      } else {
+        toc.style.display = 'none';
+      }
+      var button = document.getElementById('main-toc-button');
+      if (button.innerHTML == 'Table of Contents') {
+        button.innerHTML = 'Hide Table of Contents';
+      } else {
+        button.innerHTML = 'Table of Contents';
+      }
+    }
+    // If the page is 'index.html', show the TOC by default
+    if (window.location.pathname.endsWith('index.html')) {
```

Comment:
Could this be done in a better way which would avoid looking at the url? @heatherleaf